### PR TITLE
fix: set enabled transaction types

### LINKED ISSUE #838: bug: SelectionType doesn't regard hasMagistrateEnabled
The `SelectionType` component still includes the magistrate transaction types even if those transactions are not supported on the bridgechain (likewise with timelock transaction types after #836).

The same applies to the transaction types which can be chosen on the advanced search pages transaction type input.

### PR DESCRIPTION
**Pull request by dated**
## Summary

Closes #838.

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

**Comment by dated** [comment]
Will finish after #836 has been merged.

**Review by ItsANameToo — APPROVED** [review]

**Comment by ghost** [comment]
A member has approved this PR. A maintainer will merge this PR shortly. If it shouldn't be merged yet, please leave a comment saying so and we'll wait.

_Thank you for your contribution!_

**Comment by ghost** [comment]
Your pull request has been merged and marked as tier 4. _It will earn you $20 USD._

**Pull request merged**
